pkg/heatapi/deployment.go

```diff
@@ -159,3 +137,58 @@ func Deployment(
 	}
 	return deployment, nil
 }
+
+func createVolumeAndMounts(instance *heatv1beta1.HeatAPI) ([]corev1.Volume, []corev1.VolumeMount, error) {
```

Comment:
Ok, I think this is working now. So, ideally, I would like these to be generic enough that we can use them for each deployment without reimplementing and duplicating the code.

Moving back to draft while I work on that.

Reply:
https://github.com/openstack-k8s-operators/heat-operator/pull/503